src/envs/touch_point.py

```diff
@@ -0,0 +1,179 @@
+    magnitude of the movement is constant. The point is considered
+    touched if the distance between the center of the robot and the
+    center of the target point is less than a certain threshold, which
+    is greater than the action magnitude.
+    """
+    x_lb: ClassVar[float] = 0.0
+    x_ub: ClassVar[float] = 1.0
+    y_lb: ClassVar[float] = 0.0
+    y_ub: ClassVar[float] = 1.0
+    action_magnitude: ClassVar[float] = 0.1
+    # The target point is touched if the distance between the robot and target
+    # is less than action_magnitude * touch_multiplier.
+    touch_multiplier: ClassVar[float] = 1.5
+
+    def __init__(self) -> None:
+        super().__init__()
+        # Types
+        self._robot_type = Type("robot", ["x", "y"])
+        self._target_type = Type("target", ["x", "y"])
+        # Predicates
+        self._Touched = Predicate("Touched",
+                                  [self._robot_type, self._target_type],
+                                  self._Touched_holds)
+        # Options
+        self._MoveTo = ParameterizedOption(
+            "MoveTo",
+            types=[self._robot_type, self._target_type],
+            params_space=Box(0, 1, (0, )),
+            policy=self._MoveTo_policy,
+            initiable=lambda s, m, o, p: True,
+            terminal=self._MoveTo_terminal)
+        # Static objects (always exist no matter the settings).
+        self._robot = Object("robby", self._robot_type)
+        self._target = Object("target", self._target_type)
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "touch_point"
+
+    def simulate(self, state: State, action: Action) -> State:
+        assert self.action_space.contains(action.arr)
+        rot, = action.arr
+        x = state.get(self._robot, "x")
+        y = state.get(self._robot, "y")
+        new_x = x + np.cos(rot) * self.action_magnitude
+        new_y = y + np.sin(rot) * self.action_magnitude
+        new_x = np.clip(new_x, self.x_lb, self.x_ub)
+        new_y = np.clip(new_y, self.y_lb, self.y_ub)
+        next_state = state.copy()
+        next_state.set(self._robot, "x", new_x)
+        next_state.set(self._robot, "y", new_y)
+        return next_state
+
+    def _generate_train_tasks(self) -> List[Task]:
+        return self._get_tasks(num=CFG.num_train_tasks, rng=self._train_rng)
+
+    def _generate_test_tasks(self) -> List[Task]:
+        return self._get_tasks(num=CFG.num_test_tasks, rng=self._test_rng)
+
+    @property
+    def predicates(self) -> Set[Predicate]:
+        return {self._Touched}
+
+    @property
+    def goal_predicates(self) -> Set[Predicate]:
+        return {self._Touched}
+
+    @property
+    def types(self) -> Set[Type]:
+        return {self._robot_type, self._target_type}
+
+    @property
+    def options(self) -> Set[ParameterizedOption]:
+        return {self._MoveTo}
+
+    @property
+    def action_space(self) -> Box:
+        # An angle in radians. Wraps around to avoid boundary issues.
```

Comment:
what part of the code is handling the boundary issues of the angle?

Reply:
the move policy seems to be just -pi to pi?

Reply:
yeah nvm I don't think this is necessary after all